src/entities/position/model/get-display-positions.ts

```diff
@@ -0,0 +1,291 @@
+    }
+
+    if (asset1IsStablecoin) {
+      return getOrdersByBaseQuoteAssets(asset2, asset1);
+    }
+
+    if (asset2IsStablecoin) {
+      return getOrdersByBaseQuoteAssets(asset1, asset2);
+    }
+
+    return asset1HasPriority
+      ? getOrdersByBaseQuoteAssets(asset2, asset1)
+      : getOrdersByBaseQuoteAssets(asset1, asset2);
+  }
+
+  // Otherwise, this is simple, if asset 1 is a numeraire then we render it as the quote asset:
+  if (asset1IsNumeraire) {
+    return getOrdersByBaseQuoteAssets(asset2, asset1);
+  }
+
+  // Otherwise, if asset 2 is a numeraire we render that one as the quote asset:
+  if (asset2IsNumeraire) {
+    return getOrdersByBaseQuoteAssets(asset1, asset2);
+  }
+
+  // It's possible that neither are, which is rare, in that case we use the canonical ordering:
+  return getOrdersByBaseQuoteAssets(asset1, asset2);
+};
+
+export const getOrderValueViews = ({
+  direction,
+  baseAsset,
+  quoteAsset,
+}: {
+  direction: string;
+  baseAsset: CalculatedAsset;
+  quoteAsset: CalculatedAsset;
+}) => {
+  // We want to render two main piece of information to the user, assuming their unit of account is the quote asset:
+  // - the price i.e, the number of unit of *quote assets* necessary to obtain a unit of base asset.
+  // - the trade amount i.e, the amount of *base assets* that the position is either seeking to purchase or sell.
+  const effectivePrice = pnum(baseAsset.effectivePrice.toString(), baseAsset.exponent).toValueView(
+    quoteAsset.asset,
+  );
+  const basePrice = pnum(baseAsset.price.toString(), baseAsset.exponent).toValueView(
+    quoteAsset.asset,
+  );
+
+  // This is the trade amount (in base asset) that the position seeks to SELL or BUY.
+  const amount =
+    direction === 'Sell'
+      ? // We are selling the base asset to obtain the quote asset, so we can simply use the current reserves.
+        pnum(baseAsset.amount.toString(), baseAsset.exponent).toValueView(baseAsset.asset)
+      : // We are buying the base asset, we need to convert the quantity of quote asset that we have provisioned.
+        pnum(
+          quoteAsset.amount.times(quoteAsset.effectivePrice).toString(),
+          quoteAsset.exponent,
+        ).toValueView(baseAsset.asset);
+
+  return {
+    direction,
+    baseAsset,
+    quoteAsset,
+    amount,
+    basePrice,
+    effectivePrice,
+  };
+};
+
+export interface GetDisplayPositionsArgs {
+  positions: Map<string, Position>[] | undefined;
+  getMetadataByAssetId: GetMetadataByAssetId;
+  asset1Filter?: Metadata;
+  asset2Filter?: Metadata;
+  stateFilter?: PositionState_PositionStateEnum[];
+}
+
+/**
+ * Takes the result of `usePositions` query and adapts it to UI
+ * by filtering positions by status/metadata, mapping into a better format, and sorting them.
+ */
+export const getDisplayPositions = ({
+  positions,
+  getMetadataByAssetId,
+  asset1Filter,
+  asset2Filter,
+  stateFilter,
+}: GetDisplayPositionsArgs): DisplayPosition[] => {
```

Comment:
refactored the arguments of this function – now it accepts filters. By default, returns all positions